reactor-pool/src/main/java/reactor/pool/AbstractPool.java

```diff
@@ -122,6 +122,18 @@ public boolean isInactiveForMoreThan(Duration duration) {
 
 	// == common methods to interact with idle/pending queues ==
 
+	void applyPendingAcquireTimeout(Borrower<POOLABLE> borrower) {
+		boolean noIdle = idleSize() == 0;
+		boolean noPermits = poolConfig.allocationStrategy().estimatePermitCount() == 0;
```

Comment:
This bit of code is not synchronized in any way, right? So it can happen that following this method call, when no timeout was launched, the borrower will still not get granted a resource and will potentially exceed the timeout waiting for a resource. Am I missing some context that guarantees atomicity of this operation or exclusiveness?

Reply:
removed that code